lib/python/qmk/cli/list_keyboards.py

```diff
@@ -0,0 +1,30 @@
+"""List the keyboards currently defined within QMK
+"""
+from milc import cli
+import os
+import re
+import glob
+
+
+@cli.subcommand("List the keyboards currently defined within QMK")
+def list_keyboards(cli):
+    """List the keyboards currently defined within QMK
+    """
+
+    base_path = os.path.join(os.getcwd(), "keyboards") + os.path.sep
+    kb_path_wildcard = os.path.join(base_path, "**", "rules.mk")
+
+    # find everywhere we have rules.mk
+    paths = glob.iglob(kb_path_wildcard, recursive=True)
```

Comment:
This allows you to avoid both the regex and the lambda.

```suggestion
    paths = [path for path in glob.iglob(kb_path_wildcard, recursive=True) if 'keymaps' in path]
```